### DISCUSSION TITLE
Is there a URL for Search Engine in Youtube history?

### DISCUSSION THREAD

**Discussion by diegopds83** [qutebrowser/qutebrowser#7993] [General]
Similar to #6748 

I have this below working perfectly for standard youtube search:

```
c.url.searchengines = {
   'yt':          'https://www.youtube.com/results?search_query={}',

```

Now Youtube also allows the search for keyword in its history

I tried this one below, but it doesn't work:

```
c.url.searchengines = {
'yth':         'https://www.youtube.com/feed/history?query={}',
```

My work-around: 

1) Quickmark:
```
 yth https://www.youtube.com/feed/history
```
2) `F` + `hint` press for the search field in YouTube history.

3) Type the keyword and presses `Enter`

Thanks in advance.

**Comment by The-Compiler** [comment]
What URL do you get to after your workaround? Is the query in there? Maybe try `search_query` or `q` as well. I can't reproduce this easily as I have watch history turned off.

**Reply by diegopds83** [reply]
**_1 -  What URL do you get to after your workaround? Is the query in there?_**

Yes. For example, if I search for "qutebrowser", returns the following URL: https://www.youtube.com/feed/history?query=qutebrowser

---

**_2 - Maybe try `search_query` or `q` as well._**

I tried with these two forms: 
- `'yth':         'https://www.youtube.com/feed/history?search_query={}',`
- `'yth':         'https://www.youtube.com/feed/history?q={}',`

And both returned the following URL:

https://www.youtube.com/feed/history

---

**_3 - I can't reproduce this easily as I have watch history turned off._**

No problem. Thank you for your help so far.


---

**My hypothesis:**

1) The standard search for YouTube videos has a "universal" URL because it will look for public videos.

2) In the case of the search within the user's history, will search the user's "confidential" information, which depends on a "loggedized" account to search within it. Therefore, there must be an additional protection layer and the search is done in a different way. Therefore, the URL returned to the user is just a "mask", not serving as a search parameter. In short, this history search engine is not visible to the user and cannot be manipulated by it (Unless someone finds out a way of doing so, obviously).

What do you think?

Thanks.
